packages/astro/src/env/vite-plugin-env-virtual-mod.ts

```diff
@@ -57,11 +61,23 @@ export function astroEnvVirtualModPlugin({
 			if (id === VIRTUAL_CLIENT_MODULE_ID) {
 				return RESOLVED_VIRTUAL_CLIENT_MODULE_ID;
 			}
+			if (id === VIRTUAL_SERVER_MODULE_ID) {
+				return RESOLVED_VIRTUAL_SERVER_MODULE_ID
+			}
 		},
-		load(id) {
+		load(id, options) {
 			if (id === RESOLVED_VIRTUAL_CLIENT_MODULE_ID) {
 				return clientTemplates!.content;
 			}
+			if (id === RESOLVED_VIRTUAL_SERVER_MODULE_ID) {
+				if (options?.ssr) {
```

Comment:
I think you should use `isServerLikeOutput(options.settings.config);` where `options` is `StaticBuildOptions`.

Reply:
I think this is irrelevant here. We just don't want the module to be imported clients-side, no matter the output. Or am I missing something?

Reply:
I think using `options?.ssr` should be fine here since it's contextual to the current `load`.